src/icrar/leap-accelerate/common/Tensor3X.h

```diff
@@ -20,14 +20,33 @@
  * MA 02111 - 1307  USA
  */
 
-#include <Eigen/Core>
-//#include <unsupported/Eigen/CXX11/Tensor>
+#pragma once
+
+#include <icrar/leap-accelerate/common/eigen_3_3_beta_1_2_support.h>
+#include <eigen3/Eigen/Core>
+#include <unsupported/Eigen/CXX11/Tensor>
 
 namespace icrar
 {
     template<typename T>
-    using Tensor3X = Eigen::Matrix<Eigen::Matrix<T, Eigen::Dynamic, 1>, Eigen::Dynamic, Eigen::Dynamic>;
+    using Tensor3X = Eigen::Tensor<T, 3>;
 
-    //template<typename T>
-    //using Tensor3X = Eigen::Tensor<T, 3>;
+    template<typename T>
+    bool isApprox(const Tensor3X<T>& lhs, const Tensor3X<T>& rhs, double tolerance)
+    {
+        for(int x = 0; x < rhs.dimension(0); ++x)
+        {
+            for(int y = 0; y < rhs.dimension(1); ++y)
+            {
+                for(int z = 0; z < rhs.dimension(2); ++z)
+                {
+                    if(abs(lhs(x, y, z) - rhs(x, y, z)) > tolerance)
+                    {
+                        return false;
+                    }
+                }
+            }
+        }
+        return true;
```

Comment:
I know the main benefit of this nested loop is that it can compare a rowMajor tensor with a columnMajor, which I may need to switch between in future. I'm only using this helper for testing for now but I'll add the dimensions check and note about optimizing.